dcgm/tests/test_dcgm.py

```diff
@@ -0,0 +1,69 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+from unittest import mock
+
+import pytest
+
+from datadog_checks.dcgm import DcgmCheck
+
+
+@pytest.fixture
+def instance():
+    return {"openmetrics_endpoint": "http://localhost:9400/metrics"}
+
+
+@pytest.fixture
+def check(instance):
+    return DcgmCheck('dcgm.', {}, [instance])
+
+
+@pytest.fixture()
+def mock_metrics():
+    f_name = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'metrics.txt')
+    with open(f_name, 'r') as f:
+        text_data = f.read()
+    with mock.patch(
+        'requests.get',
+        return_value=mock.MagicMock(
+            status_code=200, iter_lines=lambda **kwargs: text_data.split("\n"), headers={'Content-Type': "text/plain"}
+        ),
+    ):
+        yield
+
+
+def test_check(dd_run_check, aggregator, check, mock_metrics):
+    dd_run_check(check)
+
+    for expected_metric in METRICS:
+        aggregator.assert_metric(
+            name=f"dcgm.{expected_metric['name']}",
+        )
```

Comment:
I think it's useful for us to add those two assertions that at some point you had commented out. The ones about all metrics and about metadata.

The one about metadata is especially relevant.

Reply:
Done ✅

Reply:
@iliakur this would mean reverting the EXPECTED_METRICS to what I have before with `"name": "dec_utilisation"`as the format will be as followed:
```
METRICS = [
    {
        "name": "dec_utilization",
        "value": 12,
        "type": AggregatorStub.MONOTONIC_COUNT,
        "tags": TAGS + ["operation:<>", "service_name:<>"],
    },
```

Reply:
Nope, [here's](https://github.com/DataDog/integrations-core/pull/14589/commits/bd3311669edd14b1772dc1d3872457c61c54c78a) what it means ;)